src/agent/debuglet.ts

```diff
@@ -426,21 +435,26 @@ export class Debuglet extends EventEmitter {
     }
 
     let project: string;
-    try {
-      project = await that.debug.authClient.getProjectId();
-    } catch (err) {
-      that.logger.error(
-        'The project ID could not be determined: ' + err.message
-      );
-      that.emit('initError', err);
-      return;
+    if (this.config.useFirebase) {
+      // Project id was discovered during initialization of the app.
+      project = (this.controller as FirebaseController).getProjectId();
+    } else {
+      try {
```

Comment:
reading the code, I don't see any reason to create a reference `that = this`, as long as you use closures of the form:

```js
() => {
 ...
}
```

`this` will reference the top level scope. I think this is probably hold over that's worth cleaning up.